java/code/src/com/redhat/rhn/frontend/xmlrpc/contentmgmt/ContentManagementHandler.java

```diff
@@ -645,6 +651,39 @@ public ContentFilter createFilter(User loggedInUser, String name, String rule, S
         }
     }
 
+    /**
+     * Create new {@link ContentFilter}s for all AppStream modules with default streams
+     *
+     * @param loggedInUser the logged in user
+     * @param prefix the filter name prefix
+     * @param channelLabel label of the modular channel
+     * @param projectLabel label of the Content Lifecycle Project
+     * @throws EntityExistsFaultException when Filter already exist
+     * @return List of created and successfully attached Filter
+     */
+    public List<ContentFilter> createAppStreamFilters(User loggedInUser, String prefix,
+            String channelLabel, String projectLabel) throws ModulemdApiException {
+        ensureOrgAdmin(loggedInUser);
+
+        Channel channel = ChannelManager.lookupByLabelAndUser(channelLabel, loggedInUser);
```

Comment:
For proper error reporting, XMLRPC methods must throw a subclass of `FaultException` when something goes wrong. 

`ChannelManager.lookupByLabelAndUser` would throw a `LookupException` if the channel does not exist. You need to catch this and throw an `EntityNotExistsFaultException` instead.

Reply:
Thanks, I tried to follow your suggestion, I moved that part into the existing try/catch block